### PR TITLE
TOAZ-221 Query landing zone based on BP id

### PR DESCRIPTION
**Pull request by sergiygetlin**
https://broadworkbench.atlassian.net/browse/TOAZ-221
Query for landing zone by billing profile id

### CODE REVIEW

**Review by zykonda — APPROVED** [review]
lgtm

File: service/src/main/java/bio/terra/workspace/db/exception/CloudContextNotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package bio.terra.workspace.db.exception;
+
+import bio.terra.common.exception.NotFoundException;
+
+public class CloudContextNotFoundException extends NotFoundException {
```
Comment (rtitle) [inline service/src/main/java/bio/terra/workspace/db/exception/CloudContextNotFoundException.java:5]: Doesn't seem like this is used in this PR?

File: service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiDispatch.java [hunk 2]
```diff
@@ -243,21 +249,26 @@ private ApiAzureLandingZoneResult toApiAzureLandingZoneResult(
...
 
-  /**
-   * TODO (https://broadworkbench.atlassian.net/browse/TOAZ-221) This method is an initial
-   * implementation that must be revised, and likely be refactored once WSM stores the LZ id in the
-   * azure context. The initial assumption is that the cardinality of 1:1 between the cloud context
-   * and the LZ is enforced in the create operation, therefore more than one LZ per azure context is
-   * not allowed.
-   */
-  public UUID getLandingZoneId(AzureCloudContext azureCloudContext) {
-    var landingZoneTarget = MapperUtils.LandingZoneTargetMapper.from(azureCloudContext);
```
Comment (rtitle) [inline service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiDispatch.java:254]: Glad to resolve this TODO.

Are we able to remove the method `MapperUtils.LandingZoneTargetMapper.from(CloudContext)`?

**Review by rtitle — APPROVED** [review]
LGTM

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storageContainer/VerifyAzureStorageContainerCanBeCreatedStep.java [hunk 3]
```diff
@@ -140,7 +140,9 @@ private StepResult validateWorkspaceStorageAccountExists(
   private StepResult validateLandingZoneSharedStorageAccountExist(
       FlightContext context, AzureCloudContext azureCloudContext, StorageManager storageManager) {
```
Comment (giventocode) [inline service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storageContainer/VerifyAzureStorageContainerCanBeCreatedStep.java:141]: After the change do we need to pass the azureCloudContext as a parameter here?

**Review by giventocode — CHANGES_REQUESTED** [review]
Looks good, a couple of questions.

File: service/src/main/java/bio/terra/workspace/db/exception/CloudContextNotFoundException.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sergiygetlin) [reply service/src/main/java/bio/terra/workspace/db/exception/CloudContextNotFoundException.java:5]: It was used before improvements. I will remove it.

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storageContainer/VerifyAzureStorageContainerCanBeCreatedStep.java (hunk 3 above)
Comment (sergiygetlin) [reply service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storageContainer/VerifyAzureStorageContainerCanBeCreatedStep.java:141]: Yes. It is used in this method

File: service/src/main/java/bio/terra/workspace/db/exception/CloudContextNotFoundException.java (hunk 1 above)
Comment (sergiygetlin) [reply service/src/main/java/bio/terra/workspace/db/exception/CloudContextNotFoundException.java:5]: I will remove it. Thanks

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storageContainer/VerifyAzureStorageContainerCanBeCreatedStep.java (hunk 3 above)
Comment (sergiygetlin) [reply service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storageContainer/VerifyAzureStorageContainerCanBeCreatedStep.java:141]: I will remove it.

File: service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiDispatch.java (hunk 2 above)
Comment (sergiygetlin) [reply service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiDispatch.java:254]: Yes. I removed it. Thanks.

**Review by giventocode — APPROVED** [review]
Thanks for making the changes.

**Pull request merged**